### PR TITLE
stm32/boards/WEACTV20_F411: Add support for WeAct v2.0 'blackpill' board

### PR DESCRIPTION
**Pull request by jeremyherbert**
The WeAct v2.0 board (also sometimes known as the "blackpill") is a micropython-capable board based on the STM32F411CE (512kB flash, 128kB SRAM) that costs only a few US dollars and is widely available from various Chinese suppliers. It is a breadboard compatible breakout-style board that has reset/BOOT0/user switches, a user LED and a USB C connector, along with 25MHz and 32.768kHz crystals.

Although the board has an unpopulated flash footprint, I don't have any SPI flash ICs around to test it and add support, so this only supports internal flash for the time being. I have ordered some and will try to test the external flash when I get a chance.

moved from https://github.com/micropython/micropython/pull/5582 to a branch

### CODE REVIEW

**Comment by mcauser** [comment]
(Board designer) YXZhu's version: https://github.com/YXZhu/micropython/tree/master/ports/stm32/boards/WeAct_F411CE

My slightly modified version: https://github.com/mcauser/WEACT_F411CEU6

Perhaps we should combine our efforts?

YXZhu also has a F401 board:
https://github.com/YXZhu/micropython/tree/master/ports/stm32/boards/WeAct_F401CC

**Comment by jeremyherbert** [comment]
Yes please! Is there a reason why @YXZhu did not create a PR for this board?

**Comment by YXZhu** [comment]
I had too many things to do and was afraid I wouldn't have time to maintain micropython so I didn't create a PR.

**Comment by jeremyherbert** [comment]
I've just updated and tested this with the external flash, it should be ready for merging now. It's based on the work linked above (and the source repos are linked in the header).

@mcauser there is an issue in your repo, for the V2.0 board the SPI flash MISO is connected to PB4.

@YXZhu is there a reason why the SPI flash was changed from PA6 to PB4?

**Comment by YXZhu** [comment]
The SDIO peripheral has a `SDIO_CMD` pin in `PA6`. If the Flash USES `PA6`, the SDIO cannot be used. Later, considering that we seldom USES the SDIO peripheral, the `V2.1+` version of Flash re-uses `PA6`

**Comment by jeremyherbert** [comment]
Ok thanks, that makes sense. I just found the schematics too: https://github.com/WeActTC/MiniF4-STM32F4x1

And I see there is now a v2.2, but it doesn’t seem to be available on aliexpress yet. I will have to keep an eye out for them.

**Comment by dpgeorge** [comment]
Is this still useful/interesting to merge?

@mcauser would it make your repo for this board obsolete?

**Comment by jeremyherbert** [comment]
I think this is still useful (I still use it), are there any changes that need to be made for it to be merged?

**Comment by dpgeorge** [comment]
> are there any changes that need to be made for it to be merged?

If you could rebase on master that would be a good start.

File: ports/stm32/boards/WEACTV20_F411/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+// I2C busses
+#define MICROPY_HW_I2C1_SCL (pin_B6)
+#define MICROPY_HW_I2C1_SDA (pin_B7)
+#define MICROPY_HW_I2C2_SCL (pin_B10)
+#define MICROPY_HW_I2C2_SDA (pin_B9)
+#define MICROPY_HW_I2C3_SCL (pin_A8)
+#define MICROPY_HW_I2C3_SDA (pin_B8)
+
+// SPI busses
```
Comment (dpgeorge) [inline ports/stm32/boards/WEACTV20_F411/mpconfigboard.h:51]: please change busses -> buses

**Pull request closed without merging**